central/declarativeconfig/transform/access_scope_test.go

```diff
@@ -0,0 +1,274 @@
+package transform
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/declarativeconfig"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestWrongConfigurationTypeTransform(t *testing.T) {
+	at := newAccessScopeTransform()
+	msgs, err := at.Transform(&declarativeconfig.AuthProvider{})
+	assert.Nil(t, msgs)
+	assert.Error(t, err)
+	assert.ErrorIs(t, err, errox.InvalidArgs)
+}
+
+func TestTransformAccessScope(t *testing.T) {
+	at := newAccessScopeTransform()
+
+	// 1. Access scope with empty rules mimicking an unrestricted scope.
+	scopeConfig := &declarativeconfig.AccessScope{
+		Name:        "test-scope",
+		Description: "test description",
+		Rules:       declarativeconfig.Rules{},
+	}
+
+	msgs, err := at.Transform(scopeConfig)
+	assert.NoError(t, err)
+	require.Len(t, msgs, 1)
+	msg := msgs[0]
+	scopeProto, ok := msg.(*storage.SimpleAccessScope)
+	require.True(t, ok)
+	assert.Equal(t, scopeConfig.Name, scopeProto.GetName())
+	assert.Equal(t, scopeConfig.Description, scopeProto.GetDescription())
+	assert.Equal(t, storage.Traits_DECLARATIVE, scopeProto.GetTraits().GetOrigin())
+	assert.Empty(t, scopeProto.GetRules().GetIncludedClusters())
+	assert.Empty(t, scopeProto.GetRules().GetNamespaceLabelSelectors())
+	assert.Empty(t, scopeProto.GetRules().GetClusterLabelSelectors())
+
+	// 2. Access scope with cluster label selectors.
+	scopeConfig = &declarativeconfig.AccessScope{
+		Name:        "test-scope",
+		Description: "test description",
+		Rules: declarativeconfig.Rules{
+			ClusterLabelSelectors: []declarativeconfig.LabelSelector{
+				{Requirements: []declarativeconfig.Requirement{
+					{
+						Key:      "a",
+						Operator: 1,
```

Comment:
Let's put something like `Operator(storage.SetBasedLabelSelector_Operator_IN)` here to be more explicit